src/Types/Output/ListOutputType.hack

```diff
@@ -32,7 +32,7 @@ final class ListOutputType<TInner> extends OutputType<vec<TInner>> {
         );
 
         foreach ($results as $idx => $result) {
-            if ($result is GraphQL\ValidFieldResult) {
+            if ($result is GraphQL\ValidFieldResult<_>) {
```

Comment:
...but that fails here. Hack correctly determines that `$result` is the intersection `FieldResult<TResolved> & ValidFieldResult<mixed>` after this check, but it can't simplify that intersection to `ValidFieldResult<TResolved>` :(